src/solver/ts-generator/thermal.cpp

```diff
@@ -606,26 +578,91 @@ std::vector<Data::ThermalCluster*> getAllClustersToGen(Data::AreaList& areas,
     return clusters;
 }
 
-bool GenerateThermalTimeSeries(Data::Study& study,
+listOfLinks getAllLinksToGen(Data::AreaList& areas)
+{
+    listOfLinks links;
+
+    areas.each([&links](Data::Area& area) {
+        std::ranges::for_each(area.links, [&links](auto& l) {
+            links.emplace_back(l.second, linkDirection::direct);
+            links.emplace_back(l.second, linkDirection::indirect);
+        });
+    });
+
+    return links;
+}
+
+void writeResultsToDisk(const Data::Study& study,
+                        Solver::IResultWriter& writer,
+                        Matrix<>& series,
+                        const std::string& savePath)
+{
+    if (study.parameters.noOutput)
+        return;
+
+    std::string buffer;
+    series.saveToBuffer(buffer, 0);
+    writer.addEntryFromBuffer(savePath, buffer);
+}
+
+bool generateThermalTimeSeries(Data::Study& study,
                                std::vector<Data::ThermalCluster*> clusters,
-                               uint year,
-                               Solver::IResultWriter& writer)
+                               Solver::IResultWriter& writer,
+                               const std::string& savePath)
 {
     logs.info();
     logs.info() << "Generating the thermal time-series";
-    Solver::Progression::Task progression(study, year, Solver::Progression::sectTSGThermal);
 
-    auto* generator = new GeneratorTempData(study, progression, writer);
+    bool archive = study.parameters.timeSeriesToArchive & Data::timeSeriesThermal;
```

Comment:
About **bool archive** : 
- no need to test its value inside the loop as it's the same for every cluster. Instead we should return if **archive** is **false**, and remove the `if (archive)` from the loop.
- Actually archive's value should be tested only if we call this function from the **solver**, but not from the **TS generator**.

In general, we should isolate code that's used in both the **TS generator** and the **solver**, and reject the other code (thus 2 pieces) elsewhere in separate places.

Reply:
It's because of `cluster->calculationOfSpinning()` at l. 621. 

It would be best to do this
```cpp
generateTimeSeries();
if (archive) 
{
    writeResultsToDisks();
}
```
But actually we have
```cpp
bool generateThermalTimeSeries()
{
    bool archive = study.parameters.timeSeriesToArchive & Data::timeSeriesThermal;
    for (auto* cluster : clusters)
    {
        if (archive) // compatibilty with in memory
        {
            writeResultsToDisk(study, writer, cluster->series.timeSeries, filePath);
        }
        cluster->calculationOfSpinning(); // SIDE EFFECT YAY
    }
    return true;
}
```

Reply:
ok, but as said in [this remark](https://github.com/AntaresSimulatorTeam/Antares_Simulator/pull/1986#discussion_r1562506106), we don't need the  calculation of spinning here.